contracts/utils/StrikeSelection.sol

```diff
@@ -5,13 +5,128 @@ pragma experimental ABIEncoderV2;
 import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
 import {IOtoken} from "../interfaces/GammaInterface.sol";
 import {DSMath} from "../vendor/DSMath.sol";
+import {IOptionsPremiumPricer} from "../interfaces/IRibbon.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
 
-contract StrikeSelection is DSMath {
+import "hardhat/console.sol";
+
+contract StrikeSelection is DSMath, Ownable {
     using SafeMath for uint256;
 
-    constructor() {}
+    /**
+     * Immutables
+     */
+    IOptionsPremiumPricer public immutable optionsPremiumPricer;
+
+    // delta for options strike price selection (ex: 1 is 100)
+    uint256 public delta;
+    // step in absolute terms at which we will increment
+    // (ex: 100 means we will move at increments of 100 points)
+    uint256 public step;
+
+    event DeltaSet(uint256 oldDelta, uint256 newDelta, address owner);
+    event StepSet(uint256 oldStep, uint256 newStep, address owner);
+
+    constructor(
+        address _optionsPremiumPricer,
+        uint256 _delta,
+        uint256 _step
+    ) {
+        require(_optionsPremiumPricer != address(0), "!_optionsPremiumPricer");
+        require(_delta > 0, "!_delta");
+        require(_step > 0, "!_step");
+        optionsPremiumPricer = IOptionsPremiumPricer(_optionsPremiumPricer);
+        // ex: delta = 10
+        delta = _delta;
+        // ex: step = 1000
+        step = _step;
+    }
+
+    /**
+     * @notice Gets the strike price satisfying the delta value
+     * given the expiry timestamp and whether option is call or put
+     * @param expiryTimestamp is the unix timestamp of expiration
+     * @param isPut is whether option is put or call
+     */
+
+    function getStrikePrice(uint256 expiryTimestamp, bool isPut)
+        external
+        view
+        returns (uint256, uint256)
+    {
+        require(
+            expiryTimestamp > block.timestamp,
+            "Expiry must be in the future!"
+        );
+
+        // asset price
+        uint256 assetPrice = optionsPremiumPricer.getUnderlyingPrice();
+
+        // For each asset prices with step of 'margin' (down if put, up if call)
+        //   if asset's getOptionDelta(currStrikePrice, t) == (isPut ? 1 - delta:delta)
+        //   with certain margin of error
+        //        return strike price
+
+        bool pastDelta = false;
+        uint256 strike = assetPrice;
```

Comment:
One easy fix to round out the strike prices to be like Deribit is to modulo the starting `strike`

```solidity
uint256 strike = assetPrice.sub(assetPrice % step);
```

So if we read 2742 as the `assetPrice`, we would set `strike = 2700` and iterate +-100 from there